src/lib/registration/remote_package.rb

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+module Registration
+  # Represents a package as seen through the SUSE::Connect API
+  #
+  # Additionally, it offers a mechanism to find out the local package (libzypp counterpart).
+  #
+  # @example Find the status
+  class RemotePackage
+    attr_reader :name, :arch, :version, :release, :addon
```

Comment:
We already have some classes dealing with `name`,  `version`, `arch` in YaST. Would it be possible to reuse them here? Or at least use as a base class?

If not then it seems we should probably create a base class for all those uses and share it...

Reply:
Having a class to handle versioning would be nice, but I would postpone it by now. We are investing too much time in this feature and I prefer to release as soon as possible.